src/NServiceBus.Core/SettingsExtensions.cs

```diff
@@ -20,16 +20,19 @@ public static T GetConfigSection<T>(this ReadOnlySettings settings) where T : cl
             var typesToScan = settings.GetAvailableTypes();
             var configurationSource = settings.Get<IConfigurationSource>();
 
-            // ReSharper disable HeapView.SlowDelegateCreation
-            var sectionOverrideType = typesToScan.Where(t => !t.IsAbstract)
-                .FirstOrDefault(t => typeof(IProvideConfiguration<T>).IsAssignableFrom(t));
-            // ReSharper restore HeapView.SlowDelegateCreation
+            var sectionOverrideTypes = typesToScan.Where(t => !t.IsAbstract && typeof(IProvideConfiguration<T>).IsAssignableFrom(t)).ToArray();
+            if (sectionOverrideTypes.Length > 1)
+            {
+                throw new Exception($"{sectionOverrideTypes.Length} configuration providers implementing IProvideConfiguration<{typeof(T).Name}> were found. Ensure that only one configuration provider per configuration section is found to resolve this error.");
```

Comment:
So, the only problem I have with this is that it's improper grammar to start a sentence with a number in numerical form.

It would be ideal if we had an easy way to convert to the word form of a number easily, i.e. "Two" instead of "2".

In the past I've used [Humanizer](https://github.com/Humanizr/Humanizer) for this, but I'm not suggesting we actually use this here.

Maybe something like `"Multiple ({sectionOverrideTypes.Length}) configuration providers... "`